src/content-tags-drawer/data/apiHooks.jsx

```diff
@@ -95,7 +84,14 @@ export const useTaxonomyTagsData = (taxonomyId, parentTag = null, numPages = 1,
     return newTags;
   }, [dataPages]);
 
-  return { hasMorePages, tagPages };
+  const flatTagPages = {
+    isLoading: tagPages.some(page => page.isLoading),
+    isError: tagPages.some(page => page.isError),
+    isSuccess: tagPages.every(page => page.isSuccess),
+    data: tagPages.flatMap(page => page.data),
+  };
+
+  return { hasMorePages, tagPages: flatTagPages };
```

Comment:
I made this change trying to simplify the component side.

Also, I'm not sure why, a new tagPages is being generated every render (I don't think this useMemo is don't its job).

Reply:
Every render. I got that when I added a useEffect with tagPages in the dependency array and updated an state in the hook. It caused a render loop. But I don't think this is a problem now.